### PR TITLE
On chain program

### PR DESCRIPTION
**Pull request by whalelephant**
Storing successful verification results onchain.

### CODE REVIEW

File: program/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[package]
+name = "eclipse-onchain-program"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+solana-program = "1.9.5"
+thiserror = "1.0.24"
+arrayref = "0.3.6"
+solana-frozen-abi = "1.9.4"
```
Comment (tuommaki) [inline program/Cargo.toml]:
```suggestion
solana-frozen-abi = "1.9.5"
```
Should all Solana deps match?

File: program/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+zeroize = "1.4.3"
+borsh = "0.9.1"
+bytemuck = { version = "1.7.2", features = ["derive"] }
+
+[dev-dependencies]
+solana-sdk = "1.9.2"
```
Comment (tuommaki) [inline program/Cargo.toml]:
```suggestion
solana-sdk = "1.9.5"
```
Same as above.

File: program/src/instruction.rs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    /// Accounts expected:
+    ///
+    /// 0. `[signer]`: The account of the account initialise the verification
+    /// 1. `[writable]`: Aleo transactions verification results storage account
+    /// 2. `[]`: Eclipse Program PDA account
+    /// 3. `[]`: Aleo Program account
+    /// 4. `[]`: System Program account
+    VerifyAleoTransaction {
+        tx_id: Vec<u8>,
+        aleo_program_id: Pubkey,
```
Comment (tuommaki) [inline program/src/instruction.rs]: Should the `aleo_program_id` be actually hard coded in the on-chain program? Doesn't this design introduce similar vulnerability as the Wormhole one?

I'm thinking that in this design a malicious person could submit own "aleo proof verifier" program ID w/ corresponding accounts and then fake that some given Aleo txs are verified.

Comment (whalelephant) [reply program/src/instruction.rs]: I agree with you here! good catch, I have moved the program_id into the onchain program.

**Review by tuommaki — APPROVED** [review]
LGTM :+1:

**Pull request merged**